samples/Shared/UnitTestFramework/DiagnosticResult.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Roslyn.UnitTestFramework
+{
+    /// <summary>
+    /// Structure that stores information about a <see cref="Diagnostic"/> appearing in a source.
+    /// </summary>
+    public struct DiagnosticResult
+    {
+        private const string DefaultPath = "Test0.cs";
+
+        private static readonly object[] EmptyArguments = new object[0];
+
+        private ImmutableArray<FileLinePositionSpan> _spans;
+        private string _message;
+
+        public DiagnosticResult(string id, DiagnosticSeverity severity)
```

Comment:
Is severity required here?

Reply:
This constructor is rarely required. Severity does not need to be specified in any of the three primary locations where this is constructed:

1. By calling `Diagnostic()`
2. By calling `CompilerError("CS0000")`
3. Starting with commit 56d48d941804cda46d4af0b7de329b03ab4149eb, by using markup syntax in the input and/or output